cpp/sarmata-client/main.cpp

```diff
@@ -137,33 +87,38 @@ po::options_description CreateOptionsDescription(void) {
              "MRCPv2 Speech-Complete-Timeout in milliseconds.")
             ("speech-incomplete-timeout", po::value<int>(),
              "MRCPv2 Speech-Incomplete-Timeout in milliseconds.")
-            ("define-grammar", "If present, will perform DefineGrammar call for given --grammar-name"
+            ("define-grammar", "If present, will perform DefineGrammar call for given --grammar-name "
              "and --grammar (will not call Recognize).");
     return optionsDescription;
 }
 
 int main(int argc, char* argv[]) {
-    using grpc::Channel;
-    using grpc::ClientContext;
-    using grpc::Status;
-    using grpc::ClientReaderWriter;
-    using namespace techmo::sarmata;
-
     try {
         po::options_description optionsDescription(CreateOptionsDescription());
         po::variables_map userOptions;
         po::store(po::command_line_parser(argc, argv).options(optionsDescription).run(), userOptions);
-        po::notify(userOptions);
-
         std::cout << "Sarmata ASR gRPC client " << LIBSARMATA_CLIENT_VERSION << std::endl;
-
         if (userOptions.count("help")) {
             std::cout << optionsDescription;
             return 0;
         }
+        po::notify(userOptions);
+
+        const techmo::sarmata::SarmataClientConfig config = CreateSarmataClientConfig(userOptions);
+        techmo::sarmata::SarmataClient sarmata_client{ userOptions["service-address"].as<std::string>() };
 
         if (userOptions.count("define-grammar")) {
-            return defineGrammar(userOptions);
+            if (config.grammar_name.empty()) {
```

Comment:
do some additional logging - which mode we are running here (defining grammar, deleting grammar)
consider extracting this as a separate method (or even add factory to create selected mode - define-grammar, run-recognition)

Reply:
I think separate methods would be enough.